libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -175,11 +175,17 @@ AP_BattMonitor::init()
 #if CONFIG_HAL_BOARD == HAL_BOARD_PX4
                 drivers[instance] = new AP_BattMonitor_SMBus_PX4(*this, instance, state[instance]);
 #else
-                drivers[instance] = new AP_BattMonitor_SMBus_I2C(*this, instance, state[instance],
+                drivers[instance] = new AP_BattMonitor_SMBus_Solo(*this, instance, state[instance],
```

Comment:
I2C is more versatile than SOLO.
I think that it is better to distinguish between SOLO and Maxcell by I2C device ID.